r/R/dataset-format.R

```diff
@@ -309,14 +309,52 @@ FileWriteOptions <- R6Class("FileWriteOptions",
   inherit = ArrowObject,
   public = list(
     update = function(table, ...) {
+      compose_err_header <- function(unsupported_passed_args) {
+        paste0(
+          oxford_paste(unsupported_passed_args),
+          ngettext(length(unsupported_passed_args),
+                   " is not a valid argument ",
+                   " are not valid arguments "),
+          "for your chosen `format`."
+        )
+      }
+
       if (self$type == "parquet") {
+        args <- list(...)
+        supported_args <- names(formals(write_parquet))
+        supported_args <- supported_args[supported_args != c("x", "sink")]
+        unsupported_passed_args <- setdiff(names(args), supported_args)
+
+        if (length(unsupported_passed_args)) {
+          err_header <- compose_err_header(unsupported_passed_args)
+          abort(err_header)
+        }
+
         dataset___ParquetFileWriteOptions__update(
           self,
           ParquetWriterProperties$create(table, ...),
           ParquetArrowWriterProperties$create(...)
         )
       } else if (self$type == "ipc") {
         args <- list(...)
+        supported_args <- c(
+          "use_legacy_format",
+          "metadata_version",
+          "codec",
+          "null_fallback"
+        )
+
+        unsupported_passed_args <- setdiff(names(args), supported_args)
+        err_info <- NULL
+
+        if (length(unsupported_passed_args)) {
+          err_header <- compose_err_header(unsupported_passed_args)
+          if("compression" %in% unsupported_passed_args) {
```

Comment:
```suggestion
          if ("compression" %in% unsupported_passed_args) {
```

Though this might be refactored if you take the suggestions up above